### DISCUSSION TITLE
TeamTalk server: Systemd starts TeamTalk too early in many configurations

### DISCUSSION THREAD

**Discussion by Nick6489** [BearWare/TeamTalk5#1994] [Q&A]
### Description

You ship a systemd unit with all TeamTalk server binaries. The following happens in every cloud configuration TeamTalk would be hosted, other than Vultr whose network's quality may have become suspect but whose management is state of the art.

### Application

- [ ] qtTeamTalk
- [ ] TeamTalkAndroid
- [ ] iTeamTalk
- [ ] TeamTalkClassic
- [X] TeamTalkServer

### Platform

- [ ] Windows
- [ ] macOS
- [ ] Android
- [ ] iOS
- [X] Linux

### Expected behavior

TeamTalk server starts on boot when enabled (systemctl enable tt5proserver)

### Actual behavior

TeamTalk Pro Server logs the following error, then closes.

`ACE_Acceptor::ACE_Acceptor: Cannot assign requested address
Unable to launch server using TCP port 10333 UDP port 10333.
Make sure the ports are not currently in use.
2023-08-20 13:06:50.827810 Data transferred - Total TX: 0 KBytes Total RX: 0 KBytes Voice TX: 0 KBytes Voice RX: 0 KBytes Desktop TX: 0 KBytes Desktop RX: 0 KBytes Media Files TX: 0 KBytes Media Files RX: 0 KBytes Video TX: 0 KBytes/sec Video RX: 0 KBytes/sec.
`

### Steps to reproduce problem

1. edit ssystemd unit file to specify location of server.
2. copy unit file to /etc/systemd/system
3. systemd enable tt5proserver
4. reboot.
I also changed the network target to network-online, as well as try a different operating system with different network management (systemd not ifupdown). I even enabled the appropriate wait one-shot services. Nothing worked. I'm using Linode, but this has happened on DigitalOcean and most other providers other than Vultr.

**Comment by bear101** [comment]
It looks like the issue is that the server cannot bind to the requested IP-address and ports. If you run `ss -tunpl` does it state that port 10333 is already in use? Otherwise check that you haven't specified an IP-address to bind to that is not available.

**Reply by Nick6489** [reply]
The IPs and ports are available. I think I forgot to inform you that if you start the server from systemd manually after boot (systemctl start tt5proserver) it comes up, no problem.
